api/handler.py

```diff
@@ -13,25 +14,22 @@ def make_conn():
     conn = None
     try:
         conn = psycopg2.connect(
-            f"dbname={DB_NAME} user={DB_USER} host={DB_HOST} password={DB_PASS}"
+            f"dbname={DB_NAME} user={DB_USER} host={DB_HOST} password={DB_PASS}  options='-c search_path={DB_SCHEMA}'"
         )
     except:
         print("I'm unable to connect to the database")
     return conn
 
 
 def postLocation(event, context):
-    # body = {
-    #     "message": "POST /location endpoint logic. Your function executed successfully!",
-    #     "input": event,
-    # }
-
-    # response = {"statusCode": 200, "body": json.dumps(body)}
+    payload = json.loads(event["body"])
+    point_str = f"'POINT({payload['longitude']} {payload['latitude']})'"
+    sql_statement = f"INSERT INTO contact_tracer.device_location (device_id,sample_date,location) VALUES ('{payload['deviceId']}', '{payload['sampleDate']}', postgis.ST_PointFromText({point_str}))"
```

Comment:
think you forgot the upsert into the device table, that's needed to have a list of devices to iterate over for score calculation

Reply:
Done